agent/bench-scripts/pbench-trafficgen

```diff
@@ -1063,214 +1151,191 @@ function inner_most_loop() {
 		cmd="$cmd --search-runtime=$iteration_search_runtime"
 		cmd="$cmd --validation-runtime=$iteration_validation_runtime"
 
-		if [ $count -ge $start_iteration_num ]; then
-			fs_unit=""
-			if [ "${loop_type}" == "default" ]; then
-				if [ "${frame_size}" -eq "${frame_size}" ] 2>/dev/null; then
-					fs_unit="B"
-				fi
-				iteration="${count}-${traffic_direction}-${frame_size}${fs_unit}-${num_flow}flows-${max_loss_pct}pct_drop"
-				record_default_iteration "${count}" "${traffic_direction}" "${frame_size}${fs_unit}" "${num_flow}flows" "${max_loss_pct}pct_drop" "$iteration"
-			elif [ "${loop_type}" == "profile" ]; then
-				tmp_traffic_profile=$(basename ${traffic_profile})
-				iteration="${count}-${tmp_traffic_profile}-${max_loss_pct}pct_drop"
-				record_profile_iteration "${count}" "${tmp_traffic_profile}" "${max_loss_pct}pct_drop" "${iteration}"
-			fi
-			tt_iteration_dir="${benchmark_run_dir}/${count}-default"
-			iteration_dir="$benchmark_run_dir/$iteration"
+		tt_iteration_dir="${benchmark_run_dir}/${count}-default"
+		iteration_dir="$benchmark_run_dir/$iteration"
+
+		if [ "${postprocess_only}" == "n" ]; then
 			mkdir -p $iteration_dir
-			benchmark_cmd_file="$iteration_dir/$benchmark_name.cmd"
-			result_stddevpct=$maxstddevpct # this test case will get a "do-over" if the stddev is not low enough
-			failures=0
-			echo "Starting iteration[$iteration] ($count of $total_iterations)"
-			log "Starting iteration[$iteration] ($count of $total_iterations)"
-
-			iteration_options=""
-			if [ $traffic_generator != "trex-txrx-profile" ]; then
-				if [ $rate == "none" ]; then
-					if [ $traffic_generator == "trex-txrx" ]; then
-						iteration_options="$iteration_options --rate-unit=% --rate=100"
-					else
-						# don't specify a rate and let binary-search.py + whatever traffic-generator figure it out
-						iteration_options="$iteration_options"
-					fi
-				else
-					iteration_options="$iteration_options --rate-unit=$rate_unit --rate=$rate"
-				fi
-			else
-				if [ "${traffic_profiles}" == "" -o "${rates}" == "none" ]; then
+
+			local iteration_options="--max-loss-pct=${max_loss_pct}"
+			if [[ ${traffic_generator} == "trex-txrx-profile" ]]; then
+				if [[ "${traffic_profile_arg}" == "" || "${rates}" == "none" ]]; then
 					iteration_options="${iteration_options} --rate-unit=% --rate=100"
 				else
 					iteration_options="${iteration_options} --rate-unit=% --rate=${rates}"
 				fi
-			fi
-			iteration_options="$iteration_options --max-loss-pct=$max_loss_pct"
-			if [ $traffic_generator == "trex-txrx-profile" ]; then
-				if [ "${traffic_profiles}" == "" ]; then
+				if [[ "${traffic_profile_arg}" == "" ]]; then
 					profile_builder_cmd_file="${iteration_dir}/profile-builder-cmd"
```

Comment:
This should really be named `profile-builder.cmd` so that the unit tests will pick it up in the gold output.